numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -251,7 +251,7 @@ npy_PyFile_Dup2(PyObject *file, char *mode, npy_off_t *orig_pos)
     if (ret == NULL) {
         return NULL;
     }
-    fd2 = PyNumber_AsSsize_t(ret, NULL);
+    fd2 = (int)PyNumber_AsSsize_t(ret, NULL);
```

Comment:
Maybe this should be
```
Py_ssize_t tmp = PyNumber_AsSsize_t(ret, NULL);
if (tmp > INT_MAX || tmp < INT_MIN) {
    /* set an error and return NULL */
}
fd2 = (int)tmp;
```

Reply:
This should also check `PyErr_Occurred`, as otherwise if `os.dup` returns non-integer nonsense (someone monkey-patched it) then you probably segfault or `SystemError`.

Reply:
added the check and added a test that monkey-patches `os.dup` to trip failures.